src/wc/wc.rs

```diff
@@ -258,21 +352,26 @@ fn print_stats(settings: &Settings, result: &Result, max_width: usize) {
     }
 }
 
-fn open(path: &str) -> StdResult<BufReader<Box<Read + 'static>>, i32> {
+fn open(path: &str) -> StdResult<InputHandle, i32> {
     if "-" == path {
-        let reader = Box::new(stdin()) as Box<Read>;
-        return Ok(BufReader::new(reader));
+        let stdin = stdin();
+        return Ok(InputHandle {
+            #[cfg(unix)]
+            file_descriptor: stdin.as_raw_fd(),
+            reader: Box::new(stdin) as Box<Read>,
```

Comment:
Do you mind refactoring the code to get rid of this `Box`?  You might be able to get away with this (without excessive changes) by making the type of reader an associated type for the `InputHandle`.